### DISCUSSION TITLE
error_regex setup difficulty

### DISCUSSION THREAD

**Discussion by JesseRMeyer** [focus-editor/focus#334] [Q&A] [answered]
Suppose I have the error message:

drive:/path/to/file.odin(30:13) Syntax Error: Expected an operand

Which I have crafted and debugged a regex string match expression (at regex101):
```^(?P<file>.*)[(](?P<line>\d+)[:](?P<col>\d+)[)] (?P<type>Syntax Error): (?P<msg>.*)```

However, if I simply supply that to error_regex directly:
```error_regex: ^(?P<file>.*)[(](?P<line>\d+)[:](?P<col>\d+)[)] (?P<type>Syntax Error): (?P<msg>.*)```

the Build Output window does not change, nor the jump to error feature doesn't appear to work.  Restarting focus didn't appear to help.

My project's config claims to be version [3] so I presume the auto-migration completed successfully.

Am I doing anything wrong? Thanks

Win10, Focus 0.3.4

**Comment by JesseRMeyer** [comment]
Interestingly the configuration above works only sometimes.  For some reason, the first two error messages are highlighted correctly and the jump-to-error functionality works.  But the last two error messages don't get special treatment.  I thought at first it may be somehow due to the fact the first two error messages only have 1 additional line of text reported beneath them after a newline, but the last error message meets that criteria.

```
F:/Play/server/src/main_windows.odin(34:15) Syntax Error: Invalid string literal
	sqlite.open(")
	            ^
F:/Play/server/src/main_windows.odin(34:16) Syntax Error: Expected a comma, got a newline
	sqlite.open(")
	            ^
F:/Play/server/src/main_windows.odin(34:17) Syntax Error: String literal not terminated
	iocp_thread_count :: 8
  iocp = win32.CreateIoCompletionPort(win ...
	^
F:/Play/server/src/main_windows.odin(36:21) Syntax Error: Expected ')', got ':'
	iocp_thread_count :: 8
```

Maybe significant is where newlines are reported:

![highlighting](https://github.com/focus-editor/focus/assets/5846309/1a193bb3-49d8-4b89-b1e2-a9c588dc8a53)

(Note I'm not creating an issue from this until I know it's not PEBAC or an inconsistency with Odin's error output, tho I've never had issues regexing errors with Sublime)

**Reply by focus-editor** [reply]
Thanks for providing example output - I'll try to debug it.

**Reply by focus-editor** [reply]
Weird, the output you provided is matched by the error regex you provided no problem 🤔 
![image](https://github.com/focus-editor/focus/assets/119373822/1e369447-5622-4e98-a055-b99419155f68)

**Reply by JesseRMeyer** [reply]
Then it may have something to do with the text string produced by Odin and ingested by Focus.  I'll fiddle around with this more on my end.  If you're interested in this more, you can grab Odin for free and modify the examples/demo/demo.odin with random syntax and hope to reproduce the issue on your end.  https://github.com/odin-lang/Odin/releases (though I am technically operating on master)

**Reply by focus-editor** [reply]
I've installed Odin and tried to reproduce it, but it all matched fine.
![image](https://github.com/focus-editor/focus/assets/119373822/def3774d-4b2d-4480-949c-9fe3c45e41e9)
However, the output doesn't look like your output - is there a command line option I need to use to display code snippets?

**Reply by JesseRMeyer** [reply]
What if you add `-show-timings -lld` to the mix?  Also, the to-be-shortly-released Odin monthly does feature changes in error reporting (so perhaps something broke).

**Reply by focus-editor** [reply]
Got one, thanks. Will investigate
![image](https://github.com/focus-editor/focus/assets/119373822/0d385b6d-85a7-4769-a4b4-0a38e28baa89)

**Reply by focus-editor** [reply]
I think I've found the bug. Will let you know when it's fixed. We might need to do another hotfix release since there's a couple more things

**Reply by JesseRMeyer** [reply]
Also -- Focus is an absolute _joy_ to use.  Thank you for your dedication and generosity in providing such a great editor for free (as well as great support on top)!

**Reply by focus-editor** [reply, marked as answer]
@JesseRMeyer please try this executable if you're still around. It did fix the error for me but another pair of eyes would help https://drive.google.com/file/d/1PWrUZV28dsrKK1_rYm6MwodmehNRM2nr/view?usp=drive_link

**Reply by JesseRMeyer** [reply]
Seems to work for me, thanks!
